### DISCUSSION TITLE
Doppler-like secrets structure

### DISCUSSION THREAD

**Discussion by artyom-p** [Infisical/infisical#812] [Q&A]
Hi, is it possible to have a base secret that can be then overridden by multiple child branches like in Doppler?

We have a multi-tenant application that requires the same set of keys for every tenant, some values are different some are not.
In Doppler, we can define a base with some standard key/values and for tenants that required different settings, we can override the base with a child key/values, that simplifies secret management a lot and also makes the overview much easier to understand.

Is something like that planned or already exists in Infisical?

**Comment by dangtony98** [comment]
Hey @artyom-p!

You can achieve this behavior with folders or path-based secret storage — you can read about it [here](https://dev.to/dangtony98/path-based-approach-to-storing-environment-variables-secrets-3e3h-temp-slug-3900722?preview=8d3b7f4af26405387082cc1bfef462f00479702a336f93fd4d3db9d1d4b924ede117914059fdfdd16ccf9dc8da37476fc1379815d1c29e162d2c49ea); [here](https://infisical.com/docs/documentation/platform/folder) are the Infisical docs for it.

Essentially, for each tenant, you can create a folder with the tenant's id; you can then store the keys for that tenant in that folder. When you do need the keys for a particular tenant, you can query for that folder or whatever path you chose to store the secrets in.

It'd be up to you to define the folder hierarchy in whatever way best fits your use-case.

**Comment by maidul98** [comment]
You can create a base set of secrets and override them with the help of folders and import feature mentioned here https://infisical.com/docs/documentation/platform/secret-reference#import-entire-folders

**Reply by artyom-p** [reply]
Hmm, that doesn't work...

tenants/
   tenant1/
   tenant2/
   tenant3/

As our applications are multitenant they need to know about all the tenants and their settings. I expect the following behavior:

token path: /tenants
fetching all items by path /tenants returns all the tenants secrets.

Actual behavior - it returns empty array.

Seems like it works only with actual secrets, not folders.

**Reply by maidul98** [reply]
are you trying to place the base secrets in the `/tenants` folder and then trying to import them into `/tenant1` `/tenant2`, etc?

**Reply by artyom-p** [reply]
We plan to do that, but in a current test case /tenants folder contains only folders, and then /tenantX containst actual keys

**Reply by artyom-p** [reply]
just for a reference, now we have yaml configs in the format:

```
tenants:
   tenant1:
      section1:
      section2:
      sectionN:
   tenant2:
      section1:
```

**Reply by maidul98** [reply]
I think it may be more easier if we chat for a bit to see how we can structure your secrets. You can sign up on my calendar for support https://calendly.com/maidull/30min?month=2023-08
